service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -38,21 +46,69 @@ public boolean allChildrenHaveFinalOrder(List<Element<Child>> children) {
         if (YES.getValue().equals(orderAppliesToAllChildren)) {
-            children.forEach(child -> child.getValue().setFinalOrderIssued(YES.getValue()));
+            children.forEach(child -> {
+                child.getValue().setFinalOrderIssued(YES.getValue());
+                child.getValue().setFinalOrderIssuedType(orderType.getLabel());
+            });
         } else {
-            List<Integer> selectedChildren = childSelector != null ? childSelector.getSelected() : new ArrayList<>();
+            List<Integer> selectedChildren;
+            if (StringUtils.isNotBlank(remainingChildIndex)) {
+                selectedChildren = List.of(Integer.parseInt(remainingChildIndex));
+            } else if (childSelector != null) {
+                selectedChildren = childSelector.getSelected();
+            } else {
+                selectedChildren =  new ArrayList<>();
+            }
             for (int i = 0; i < children.size(); i++) {
                 Child child = children.get(i).getValue();
                 if (!selectedChildren.isEmpty() && selectedChildren.contains(i)) {
                     child.setFinalOrderIssued(YES.getValue());
+                    child.setFinalOrderIssuedType(orderType.getLabel());
                 } else if (StringUtils.isEmpty(child.getFinalOrderIssued())) {
                     child.setFinalOrderIssued(NO.getValue());
                 }
             }
         }
         return children;
     }
+
+    public Optional<Integer> getRemainingChildIndex(List<Element<Child>> allChildren) {
+        Optional<Integer> remainingChildIndex = Optional.empty();
+        for (int i = 0; i < allChildren.size(); i++) {
+            if (!YES.getValue().equals(allChildren.get(i).getValue().getFinalOrderIssued())) {
+                if (remainingChildIndex.isEmpty()) {
+                    remainingChildIndex = Optional.of(i);
+                } else {
+                    return Optional.empty();
+                }
+            }
+        }
+
+        return remainingChildIndex;
+    }
+
+    public String getRemainingChildren(List<Element<Child>> allChildren) {
+        return allChildren.stream()
+            .filter(child -> !YES.getValue().equals(child.getValue().getFinalOrderIssued()))
+            .map(child -> child.getValue().getParty().getFullName())
+            .collect(Collectors.joining("\n"));
+    }
+
+    public String getFinalOrderIssuedChildren(List<Element<Child>> allChildren) {
+        return allChildren.stream()
+            .filter(child -> YES.getValue().equals(child.getValue().getFinalOrderIssued()))
+            .map(child -> {
+                StringBuilder builder = new StringBuilder();
+                builder.append(child.getValue().getParty().getFullName());
+                if (child.getValue().getFinalOrderIssuedType() != null) {
+                    builder.append(String.format(" - %s issued", child.getValue().getFinalOrderIssuedType()));
+
+                }
```

Comment:
Seeing as this is being used twice, thoughts on have this be a method in the child class, maybe asLabel or something like that?